src/providers/conversion/Coingecko.ts

```diff
@@ -20,14 +18,19 @@ export class Coingecko implements ConversionRateProvider {
     }
   }
 
+  async load(fromIds: string, toIds: string): Promise<Record<string, Record<string, number>>> {
+    const url = `https://api.coingecko.com/api/v3/simple/price?ids=${fromIds}&vs_currencies=${toIds}`;
+    const response = await axios(url);
+    const data = await response.data;
+    return data;
+  }
+
   async getRate(from: Currency | Currency[], to: Currency | Currency[]) {
     const fromArray = typeof from === 'string' ? [from] : from;
     const toArray = typeof to === 'string' ? [to] : to;
     const fromIds = fromArray.map((currency) => Coingecko.translateCurrency(currency)).join(',');
     const toIds = toArray.map((currency) => Coingecko.translateCurrency(currency)).join(',');
-    const url = `https://api.coingecko.com/api/v3/simple/price?ids=${fromIds}&vs_currencies=${toIds}`;
-    const response = await axios(url);
-    const data = await response.data;
+    const data = await this.load(fromIds, toIds);
     return fromArray.reduce((previousPairs, fromCurrency) => {
```

Comment:
To remove typecasting:
```suggestion
    return fromArray.reduce<ConversionRatePair[]>((previousPairs, fromCurrency) => {
```

Reply:
Sure